orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/ServerGroupMetadataTagTask.groovy

```diff
@@ -16,80 +16,13 @@
 
-import java.util.concurrent.TimeUnit
-import com.fasterxml.jackson.annotation.JsonProperty
-import com.netflix.spinnaker.orca.ExecutionStatus
-import com.netflix.spinnaker.orca.RetryableTask
-import com.netflix.spinnaker.orca.TaskResult
-import com.netflix.spinnaker.orca.clouddriver.KatoService
-import com.netflix.spinnaker.orca.clouddriver.model.TaskId
-import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask
-import com.netflix.spinnaker.orca.pipeline.model.Stage
 import groovy.util.logging.Slf4j
-import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 
 @Component
 @Slf4j
-class ServerGroupMetadataTagTask extends AbstractCloudProviderAwareTask implements RetryableTask {
-  long backoffPeriod = TimeUnit.SECONDS.toMillis(5)
-  long timeout = TimeUnit.MINUTES.toMillis(5)
-
-  @Autowired
-  KatoService kato
-
-  @Autowired
-  Collection<ServerGroupEntityTagGenerator> tagGenerators
-
-  @Override
-  TaskResult execute(Stage stage) {
-    try {
-      List<Map> tagOperations = buildTagOperations(stage)
-      if (!tagOperations) {
-        return new TaskResult(ExecutionStatus.SKIPPED)
-      }
-      TaskId taskId = kato.requestOperations(tagOperations).toBlocking().first()
-      return new TaskResult(ExecutionStatus.SUCCEEDED, new HashMap<String, Object>() {
-        {
-          put("notification.type", "upsertentitytags")
-          put("kato.last.task.id", taskId)
-        }
-      })
-    } catch (Exception e) {
-      log.error("Failed to tag deployed server groups (stageId: ${stage.id}, executionId: ${stage.execution.id})", e)
-      return new TaskResult(ExecutionStatus.FAILED_CONTINUE)
-    }
-  }
-
-  private List<Map> buildTagOperations(Stage stage) {
-    def operations = []
-    ((StageData) stage.mapTo(StageData)).deployServerGroups.each { String region, Set<String> serverGroups ->
-      serverGroups.each { String serverGroup ->
-        Collection<Map<String, Object>> tags = tagGenerators ? tagGenerators.findResults { it.generateTag(stage) } : []
-        if (!tags) {
-          return []
-        }
-        operations <<
-          [
-            "upsertEntityTags": [
-              tags     : tags,
-              entityRef: [
-                entityType   : "servergroup",
-                entityId     : serverGroup,
-                account      : getCredentials(stage),
-                region       : region,
-                cloudProvider: getCloudProvider(stage)
-              ]
-            ]
-          ]
-      }
-    }
-
-    return operations
-  }
-
-  static class StageData {
-    @JsonProperty("deploy.server.groups")
-    Map<String, Set<String>> deployServerGroups = [:]
-  }
+@Deprecated
+class ServerGroupMetadataTagTask extends AddServerGroupEntityTagsTask {
+  // TODO: Remove after 11/11/17 - only here to smooth over initial round of deployments
+  // Just use AddServerGroupEntityTagsTask
```

Comment:
@ajordens @robfletcher doing this to address concerns about deployment and changing the class name of the task